pkg/query/api/v1.go

```diff
@@ -203,13 +206,18 @@ func (api *API) parseEnableDedupParam(r *http.Request) (enableDeduplication bool
 }
 
 func (api *API) parseDownsamplingParamMillis(r *http.Request, step time.Duration) (maxResolutionMillis int64, _ *ApiError) {
+	// If no max_source_resolution is specified fit at least 5 samples between steps.
+	return api.parseDownsamplingParamMillisWithDefault(r, step/5)
```

Comment:
Consider removing this function, and just calling `api.parseDownsamplingParamMillisWithDefault(r, step/5)` directly.